test/distributed/tensor/test_matrix_ops.py

```diff
@@ -33,8 +33,16 @@ def scale_for_fp8(
         t = t.unsqueeze(0).unsqueeze(-2)
     else:
         t = t.unflatten(0, (scale_shape[0], -1)).unflatten(-1, (scale_shape[1], -1))
-    scale = t.abs().amax(dim=[1, -1]).float() / torch.finfo(torch.float8_e4m3fn).max
-    t_fp8 = (t / scale[:, None, :, None]).to(torch.float8_e4m3fn)
+
+    if TEST_WITH_ROCM:
+        scale = (
+            t.abs().amax(dim=[1, -1]).float() / torch.finfo(torch.float8_e4m3fnuz).max
+        )
+        t_fp8 = (t / scale[:, None, :, None]).to(torch.float8_e4m3fnuz)
+    else:
+        scale = t.abs().amax(dim=[1, -1]).float() / torch.finfo(torch.float8_e4m3fn).max
+        t_fp8 = (t / scale[:, None, :, None]).to(torch.float8_e4m3fn)
+
```

Comment:
Please see how we handle ocp vs fnuz in test_matmul_cuda.py.  Now that we have this other test file that needs the same logic, we need to find a place to put it under torch/testing/_internal that can be imported into both this test__matrix_ops.py and test_matmul_cuda.py.

copy/paste from test_matmul_cuda.py
```
if torch.version.hip and 'gfx94' in torch.cuda.get_device_properties(0).gcnArchName:
    e4m3_type = torch.float8_e4m3fnuz
    e5m2_type = torch.float8_e5m2fnuz
    E4M3_MAX_POS = torch.finfo(torch.float8_e4m3fnuz).max
    E5M2_MAX_POS = torch.finfo(torch.float8_e5m2fnuz).max
else:
    e4m3_type = torch.float8_e4m3fn
    e5m2_type = torch.float8_e5m2
    E4M3_MAX_POS = torch.finfo(torch.float8_e4m3fn).max
    E5M2_MAX_POS = torch.finfo(torch.float8_e5m2).max
```